src/components/NcTextField/NcTextField.vue

```diff
@@ -130,10 +132,9 @@ export default {
 </docs>
 
 <template>
-	<NcInputField v-bind="{...$attrs, ...$props }"
+	<NcInputField v-bind="$props"
 		ref="inputField"
 		:trailing-button-label="clearTextLabel"
-		v-on="$listeners"
 		@input="handleInput">
 		<!-- Default slot for the leading icon -->
 		<slot />
```

Comment:
Because we use the `default` slot to pass an icon and shorthand syntax is used here, all the content described between `<NcInputField>` tags is always passed to the default slot (including empty lines, comments, and etc.). So `NcInputField` always receives and renders the default slot.

But if we use full syntax to pass only `<slot />` to the default `slot`, we can pass through the default slot only if a wrapper also has a slot content.

```suggestion
		<template v-if="$slots.default" #default>
			<slot />
		</template>
```

Reply:
But then we also need to put the icons in a template tag with a `v-if`.

So instead of this:
```vue
<NcTextField value="Text">
	<Icon1 v-if="icon1" />
	<Icon2 v-else-if="icon2" />
	<Icon3 v-else-if="icon3" />
</NcTextField>
```
one would need
```vue
<NcTextField value="Text">
	<template v-if="icon1 || icon2 || icon3" #default>
		<Icon1 v-if="icon1" />
		<Icon2 v-else-if="icon2" />
		<Icon3 v-else-if="icon3" />
	</template>
</NcTextField>
```
no?

I think I don't like this. It seems to add a lot of duplicate code and introduce an unnecessary breaking change.

Reply:
> no?

Yes, in this case, where there can be a number of possible icons and at the same time there could be no icon at all - it won't work without such v-if.

```vue
<!-- Works, always has content -->
<NcTextField value="Text">
	<Icon1 v-if="icon1" />
	<Icon2 v-else-if="icon2" />
	<Icon3 v-else="icon3" />
</NcTextField>

<!-- Works, pass content if needed -->
<!-- Doesn't work with the current solution because `computed` doesn't react on `$slots`, only on render function inside it -->
<NcTextField value="Text">
	<template v-if="icon1" #default>
		<Icon1 />
	</template>
</NcTextField>

<!-- Doesn't work, but it's Vue 3's breaking change -->
<NcTextField value="Text">
	<Icon1 v-if="icon1" />
</NcTextField>

<!-- Doesn't work without "v-if="icon1 || icon2 || icon3" -->
<NcTextField value="Text">
	<template #default>
		<Icon1 v-if="icon1" />
		<Icon2 v-else-if="icon2" />
		<Icon3 v-else-if="icon3" />
		<!-- else no icon -->
	</template>
</NcTextField>
```

Reply:
I feel like this reactivity caveat will get back to us at some point. But I guess we can fix this later, as it is an general vue3 issue, if I understood correctly.

@ShGKme Is there a different solution to this problem then migrating to a render function? Seems a little overkill to me. Maybe don't make it a computed, but just use `$slots.default` in the template directly?

Reply:
At least this works:
```vue
<NcTextField v-model:value="text1">
	<Magnify v-if="text1" />
</NcTextField>
```
although it's rather a quirk/side effect than actually intended.

Reply:
I see 4 solutions here:
1. Just check for `!!$slots.default` in a template/method
   - It doesn't work for passing empty content like your example with 3 icons or when `v-if` is used on content and not on `<template v-slot>`
2. Call `isStopPopulated()` in template/method (the same effect as re-writing to render function)
   - Works in any case
   - A bit not great in performance, slot is rendered twice. But it should not be a problem for just an icon slot
3. Keep the current solution:
   - Slot is rendered twice only when changed
   - No reaction when `v-if` is used on `<template v-slot>`, but we can combine `isSlotPopulated` with `!!$slots.default` check in the template
4. Overkill solution - save `vnodes` from `slotVnodes = $slots.default()` and then reuse them - pass to `isSlotPopulated(slotVnodes)` and render as `<component :is="slotVnodes">` instead of `<slot>`
   - Solves all problems
   - Overkill 🙈
   
No is great for me :D

Speaking about only this specific component, I'd say that #1 is fine just because I don't see a case for `NcTextField` to dynamically add and remove an icon. It would move all the input content with padding, and it seems that it is usually static.

But if I'm wrong and there is a case, then I'm totally fine with using `isSlotPopulated()` as it is used in your solution (and maybe + check `!!$slots.default` in the template just in case).

For components with a large slot content, I'd go with the 4th solution to avoid this (possibly small) performance issue.

Reply:
I still think that # 1 is not an option as it forces devs to wrap the default slot content in a `<template>` and while for sure correct, I have never seen anyone doing so by default.
# 2 might be ok here, since it is a small component and the slot usually contains an icon only, which should be cheap to render.
# 3 is also fine with me. We can still adjust if it turns out to be an issue.
# 4 probably also fine, but as you said, a bit of an overkill here.

Honestly, I would tend towards # 3 at the moment, keep the issue in mind, but move on for now and fix it if it really becomes a problem.